packages/chain-adapters/src/utxo/UtxoBaseAdapter.ts

```diff
@@ -460,61 +463,93 @@ export abstract class UtxoBaseAdapter<T extends UtxoChainId> implements IChainAd
       cursor: input.cursor,
     })
 
-    const getAddresses = (tx: unchained.utxo.types.Tx): string[] => {
-      const addresses: string[] = []
+    const transactions: Transaction[] = []
+    for (const tx of data.txs ?? []) {
+      const {
+        ownedAddresses,
+        ownedSendAddresses,
+        unownedReceiveAddresses,
+        receiveAddresses,
+        sendAddresses,
+        ownedReceiveAddresses,
+      } = getAddresses(tx, this.accountAddresses[input.pubkey])
+
+      // a send transaction where all outputs are sent to owned addresses
+      const isSelfSend =
+        ownedSendAddresses.length &&
+        receiveAddresses.every(address => ownedAddresses.includes(address))
+
+      let transaction = {} as Omit<Transaction, 'transfers'>
+      const transfers: Record<TransferType, TxTransfer[]> = { Contract: [], Send: [], Receive: [] }
+      for (const address of ownedAddresses) {
+        const parsedTx = await this.parser.parse(tx, address)
+
+        // create transaction object with all shared properties
+        if (!Object.keys(transaction).length) {
+          transaction = {
+            address: input.pubkey,
+            blockHash: parsedTx.blockHash,
+            blockHeight: parsedTx.blockHeight,
+            blockTime: parsedTx.blockTime,
+            chainId: parsedTx.chainId,
+            confirmations: parsedTx.confirmations,
+            status: parsedTx.status,
+            trade: parsedTx.trade,
+            txid: parsedTx.txid,
+          }
+        }
 
-      tx.vin?.forEach(vin => {
-        if (!vin.addresses) return
-        addresses.push(...vin.addresses)
-      })
+        // add fee if it exists on any of the parsed transactions
+        if (parsedTx.fee) transaction.fee = parsedTx.fee
```

Comment:
Does this mean this will only display the fee of the last found Tx for a grouped transaction?

Reply:
a single transaction will result in n parsed transactions (one for each address). If your account belongs to any of the send addresses you will be responsible for the fee so set it. I just didn't want one of the receive parsed transactions to overwrite the fee with `undefined`, but I can move this bit to the `transfer.type === 'Send` condition to have it make more sense. The fee will be the total transaction fee.

there is a whole separate tx scenario of multiple wallets constructing a send transaction that none of this logic accounts for, but there should be little to no transactions like this especially for user transactions when interacting through our platform.